### PR TITLE
Get tests working correctly on ubuntu-latest platform

### PR DESCRIPTION
**Pull request by aarsilv**
Our tests were not running correctly when the GitHub action platform was `ubuntu-latest`. This is because we did not specify a version or the `maven-surefire-plugin`.  Without a version specified, it defaults to a plugin version specified in the Maven Super POM that is bundled with Maven. My suspicion is that the versions differed between `osx-latest` and `ubuntu-latest`. To get our tests running correctly, wespecify the version. 

Also, we want our SDK to be compatible with Java 8. To ensure this, we run our tests in a Java 8 environment, so we need our test libraries to be compatible with Java 8.

According tot he [Mockito GitHub page](https://github.com/mockito/mockito?tab=readme-ov-file):
> [Mockito 5](https://github.com/mockito/mockito/releases/tag/v5.0.0) switches the default mockmaker to mockito-inline, and now requires Java 11

Therefore, we bump our `mockito-core` version from 5 to 4.

### CODE REVIEW

File: .github/workflows/lint-test-sdk.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 
 jobs:
   lint-test-sdk:
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
```
Comment (aarsilv) [inline .github/workflows/lint-test-sdk.yml:10]: We now match our `publish-sdk.yml` action 📈

File: pom.xml [hunk 2]
```diff
@@ -44,6 +44,7 @@
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-surefire-plugin</artifactId>
+                        <version>3.2.5</version>
```
Comment (aarsilv) [inline pom.xml:47]: The fix to get the tests to run properly

File: pom.xml [hunk 3]
```diff
@@ -120,20 +122,20 @@
         </dependency>
         <dependency>
             <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter</artifactId>
+            <artifactId>junit-jupiter-api</artifactId>
```
Comment (aarsilv) [inline pom.xml:125]: The more correct way to import JUnit5 (may have helped too 🤷‍♂️ )

File: pom.xml [hunk 4]
```diff
@@ -120,20 +122,20 @@
...
             <version>5.10.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-all</artifactId>
-            <version>1.9.5</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
```
Comment (aarsilv) [inline pom.xml:131]: The more correct way to support JUnit5's `@ParameterizedTest` (may have helped too 🤷‍♂️ )

File: pom.xml [hunk 5]
```diff
@@ -120,20 +122,20 @@
...
+            <version>5.10.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>5.10.0</version>
+            <version>4.11.0</version>
```
Comment (aarsilv) [inline pom.xml:138]: Use a version of Mockito we know is compatible with Java 8 (may have helped too 🤷‍♂️ )

File: src/test/java/com/eppo/sdk/EppoClientTest.java [hunk 6]
```diff
@@ -183,7 +181,7 @@ public void logAssignment(AssignmentLogData logData) {
             anyString(),
             anyString(),
             any(EppoAttributes.class),
-            anyMapOf(String.class, EppoAttributes.class)
+            anyMap()
```
Comment (aarsilv) [inline src/test/java/com/eppo/sdk/EppoClientTest.java:184]: Slight change needed for lower version of Mockito

File: src/test/java/com/eppo/sdk/EppoClientTest.java [hunk 7]
```diff
@@ -222,7 +220,7 @@ public void logAssignment(AssignmentLogData logData) {
             anyString(),
             anyString(),
             any(EppoAttributes.class),
-            anyMapOf(String.class, EppoAttributes.class)
+            any()
```
Comment (aarsilv) [inline src/test/java/com/eppo/sdk/EppoClientTest.java:223]: Another change needed for lower version of Mockito. `any()` used here so `nulls` are supported too--needed for the graceful error mode test.

**Review by leoromanovsky — APPROVED** [review]

**Pull request merged**